lib/client-assets.php

```diff
@@ -707,7 +707,7 @@ function gutenberg_extend_block_editor_settings_with_fse_theme_flag( $settings )
 /**
  * Sets the editor styles to be consumed by JS.
  */
-function gutenberg_extend_block_editor_styles_html() {
+function gutenberg_add_iframed_editor_assets_html() {
```

Comment:
@gziolo How would this be named in WP core?
@mcsf Is this a good name? :)

Reply:
Is the mention of iframe too implementation-specific? Ultimately, the function returns nothing and has the side effect of `echo "<script>window.__editorAssets = $editor_assets</script>";`, so I'd favour something as close to the original as `gutenberg_extend_block_editor_assets` — or alternatively along the lines of `gutenberg_prepare_block_editor_frontend_assets`?

Also, don't forget to update the doc comment for the function!

Reply:
It's not for the frontend, it's for the editor. It's also very specific for the use in the iframe and ideally we need to refactor this to not output HTML but hopefully the sources formatted as JSON.

Reply:
> It's not for the frontend, it's for the editor

Right, that's the reason for my poor attempt with `_block_editor_frontend_` . But since some of the assets will be editor-only, it's no good.

> It's also very specific for the use in the iframe and ideally we need to refactor this to not output HTML but hopefully the sources formatted as JSON.

Hm, that's fair. I'm ok with your initial suggestion, then. :)

Reply:
It's a hard one to name properly, mayby `gutenberg_share_block_editor_assets_with_iframe'. It's a callback for the filter so in WP core we can replace `gutenberg_` with `wp_`. At least this is the most popular handling. The challenge here is also to ensure it doesn't get printed twice, so in Gutenberg we should either bail out early when the function is defined in core or maybe better to remove the filter that that WP core registers.

Reply:
What should we do now that this has landed in core? Should we remove the core filter when adding this in Gutenberg?

Reply:
If we want to override the filter that exists in WP core we should remove the original one, before adding a newer version. If the version present in WP core is good enough we would need something to bridge WP 5.7 anyway. We can use an action name that is in WP 5.8 as a way to perform feature detection.

Reply:
What we currently do in Gutenberg, does it not output the scripts twice? So we'll have to do it.